server-slicer/utils.py

```diff
@@ -0,0 +1,1040 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import base64
+import itertools
+import json
+import logging
+import os
+import re
+import tempfile
+from copy import deepcopy
+from io import BytesIO
+from pathlib import Path
+from shutil import copyfile, move, rmtree
+from uuid import uuid4
+from zipfile import ZipFile
+
+import nibabel as nib
+import numpy as np
+import requests
+import skimage
+from monai.transforms import Compose, LoadImageD, MapTransform, OrientationD, ScaleIntensityD, ScaleIntensityRangeD
+from PIL import Image
+from PIL import Image as PILImage
+from PIL.Image import Image
+from tqdm import tqdm
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+logger.addHandler(logging.StreamHandler())
+
+
+SYS_PROMPT = None  # only useful in local mode
+
+REMOTE_URL = "https://developer.download.nvidia.com/assets/Clara/monai/samples"
+
+MODEL_CARDS = (
```

Comment:
I think some variables, classes and functions in this script are exactly the same as the ones in [utils.py](https://github.com/Project-MONAI/VLM/blob/main/m3/demo/experts/utils.py)

Can we include the `utils.py` in the PYTHON path and import them to avoid duplicate code? Thanks!

Reply:
Good point, @mingxin-zheng. We should actually unified this into a single folder? Should i move this to the demo folder?

Reply:
True - maybe we should make a general util folder instead of copy-pasting snippet between folders.  Can you add a `TODO: unify this class with demo/experts/utils.py" and we can check this later?